src/Kestrel.Core/Internal/Http2/Http2Stream.cs

```diff
@@ -188,6 +215,21 @@ private void AbortCore(ConnectionAbortedException abortReason)
 
             // Unblock the request body.
             RequestBodyPipe.Writer.Complete(new IOException(CoreStrings.Http2StreamAborted, abortReason));
+
+            // Stop counting any buffered bytes for this stream towards the connection input flow-control window.
+            _inputFlowControl.Abort();
```

Comment:
This needs to update the connection flow control.

Reply:
It does internally.